### PR TITLE
[shopsys] ckeditor is now inicialized after click

### LINKED ISSUE #1852: Multiple ckeditors on page slows page rendering significantly
### What is happening
Each ckeditor field renders its own javascript and initializes the component.
When there are many ckeditor fields on the page (10 domains and another rich field renders 20 ckeditors on the product detail) page rendering is really slow.

Maybe this https://symfony.com/doc/current/bundles/FOSCKEditorBundle/usage/loading.html can help.

### Expected result
Page rendering is not affected by number of ckeditor fields on the page (that much :))

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Rendering of CKEditor can slow down page load. Now the CKEditor is initialized after click to the appropriate field. Also, the field preview is rendered in admin when CKEditor is not initialized
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| closes #1852 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/assets/js/admin/components/CkEditorPreview.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Register from '../../common/utils/Register';
+import Translator from 'bazinga-translator';
+
+export default class CkEditorPreview {
+
+    constructor ($ckEditorPreview) {
+        const expandText = Translator.trans('... expand ...');
```
Comment (TomasLudvik) [inline packages/framework/assets/js/admin/components/CkEditorPreview.js:7]:
```suggestion
        const expandText = Translator.trans('Click here to edit this text');
```

and use this for all CKEditors not only with longer texts?

File: packages/framework/assets/styles/admin/todo.less [hunk 2]
```diff
@@ -2,3 +2,61 @@
...
+        }
+    }
+
+    textarea {
+        display: none;
+    }
+
+    &-expand-button {
+        display: none;
+        text-align: center;
```
Comment (TomasLudvik) [inline packages/framework/assets/styles/admin/todo.less:43]:
```suggestion
        text-align: left;
        padding-left: 15px;
```

File: packages/framework/assets/styles/admin/todo.less [hunk 3]
```diff
@@ -2,3 +2,61 @@
...
+        width: 100%;
+        height: @expand-button-height;
+        line-height: @expand-button-height;
+        background-color: @bg-info;
```
Comment (TomasLudvik) [inline packages/framework/assets/styles/admin/todo.less:47]:
```suggestion
        background-color: #ffda6b;
```

Comment (TomasLudvik) [reply packages/framework/assets/styles/admin/todo.less:47]: This color is not flashing in the eyes but catches your look in order to tell you - Hey mate, if you wanna play with this CKEditor, click me! :)

File: packages/framework/src/Component/CkEditor/CkEditorRendererDecorator.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\CkEditor;
+
+use FOS\CKEditorBundle\Renderer\CKEditorRendererInterface;
+
+class CkEditorRendererDecorator implements CKEditorRendererInterface
```
Comment (TomasLudvik) [inline packages/framework/src/Component/CkEditor/CkEditorRendererDecorator.php:9]:
```suggestion
class CKEditorRendererDecorator implements CKEditorRendererInterface
```

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 5]
```diff
@@ -34,6 +34,9 @@ msgstr "-- Vyberte značku --"
 msgid "-- Choose unit --"
 msgstr "-- Vyberte jednotku --"
 
+msgid "... expand ..."
+msgstr ""
+
```
Comment (TomasLudvik) [inline packages/framework/src/Resources/translations/messages.cs.po:39]: Missing czech translations

File: packages/framework/assets/js/admin/components/CkEditorPreview.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        const collapseText = Translator.trans('collapse');
```
Comment (TomasLudvik) [inline packages/framework/assets/js/admin/components/CkEditorPreview.js:8]: This is never displayed

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: packages/framework/assets/js/admin/components/CkEditorPreview.js (hunk 6 above)
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js:8]: Actually, it is. 
The longer text is collapsed by default and can be expanded with '...expand...' button. When text is expanded (ckeditor is not initialized), this text is rendered at the bottom of the expanded field and allows to collapse the long text back.

File: packages/framework/assets/js/admin/components/CkEditorPreview.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+        $ckEditorPreview.click(() => {
+            $ckEditorPreview.hide();
+            $expandButton.hide();
+        });
+
+        $expandButton.click(() => {
+            $ckEditorPreview.toggleClass('cke-preview-text-collapsed');
+            $expandButton.toggleClass('cke-preview-expand-button-collapsed');
+            this.toggleText($expandButton, expandText, collapseText);
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js:24]: This allows us to expand and collapse long text. Do you think it's not necessary and only expand the preview is enough?

File: packages/framework/assets/styles/admin/todo.less (hunk 3 above)
Comment (grossmannmartin) [reply packages/framework/assets/styles/admin/todo.less:48]: This button should not initialize ckeditor, but merely expand the preview. 
Is it different for you?

File: packages/framework/assets/js/admin/components/CkEditorPreview.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import Register from '../../common/utils/Register';
+import Translator from 'bazinga-translator';
+
+export default class CkEditorPreview {
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js]: changed in d0f068f63

File: packages/framework/assets/js/admin/components/CkEditorPreview.js (hunk 1 above)
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js]: I'm not sure about the best approach. I would like to wait for the UX guy's opinion as we planned.

File: packages/framework/src/Component/CkEditor/CkEditorRendererDecorator.php (hunk 4 above)
Comment (grossmannmartin) [reply packages/framework/src/Component/CkEditor/CkEditorRendererDecorator.php]: fixed in d0f068f63

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 5 above)
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po]: 😱
Thanks for noticing! fixed in b57728196

File: packages/framework/assets/js/admin/components/CkEditorPreview.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js]: Reworked, so now only edit button is present (expand preview text functionality was removed)

File: packages/framework/assets/js/admin/components/CkEditorPreview.js (hunk 1 above)
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js]: Reworked, so now only edit button is present (expand preview text functionality was removed)

File: packages/framework/assets/styles/admin/todo.less (hunk 3 above)
Comment (grossmannmartin) [reply packages/framework/assets/styles/admin/todo.less:52]: I leave it to @Srnka392 to decide the best colors

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 5 above)
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po]: translations no longer necessary

File: packages/framework/assets/js/admin/components/CkEditorPreview.js (hunk 6 above)
Comment (grossmannmartin) [reply packages/framework/assets/js/admin/components/CkEditorPreview.js]: Reworked, so now only edit button is present (expand preview text functionality was removed)

File: packages/framework/assets/styles/admin/todo.less (hunk 3 above)
Comment (Srnka392) [reply packages/framework/assets/styles/admin/todo.less]: I used for this button `btn--primary` modification, which is the blue color version of the button

**Review by TomasLudvik — APPROVED** [review]

**Review by sslt — APPROVED** [review]

**Pull request merged**